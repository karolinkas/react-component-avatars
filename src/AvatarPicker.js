import React, { Component } from 'react';

import {Image} from "./Image.js"

export class AvatarPicker extends Component {
    constructor (props){
      super(props);
      this.state = {
        imageData: props.imageData,
        selected: 0
      };
      this.getCurrentImage = (i) => {
        if (this.state.imageData[i] && this.state.imageData[i].src){
          this.setState({selected: i + 1});
        }
      }
    }
  
    grid () { 
      const grid = [];
      this.gridElements = this.imageList();
      
      //this allows us to easily change the design by just changing the columnNumber
      const columnNum = 4;
      let rowCount = 0;
      for (let i = 0; i < this.gridElements.length; i += columnNum) {
            const oneRow = [];
            rowCount ++;
            oneRow.push(this.gridElements.slice(i, i + columnNum).map(item => {
          return <div key={`column${item.props.image.label} + row${rowCount}`} className="gridElement" style={{display: 'inline-block'}}>{item}</div>
      }))
      grid.push(oneRow.map(item => {return <div key={item.toString()}>{item}</div>}))
      }
      return grid;
    }
  
  
    imageList (){
      return this.state.imageData.map((image, i) => {
        return (<Image image={image} imageToGrid={() => this.getCurrentImage(i)}/>
        );
      });
    }
    
    render (){
        console.log(this);
        return (
            <div onClick={() => this.props.pickerToApp(this.state.selected)} className="popover">
                {this.grid()}
            </div>
      );
    }
}



