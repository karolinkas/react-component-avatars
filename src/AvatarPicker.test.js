import React from 'react';
import ReactDOM from 'react-dom';

import AvatarPicker from './AvatarPicker';

import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const exampleData =  {
    imageData: [{
      src: "niceFolder",
      label: 1
    }]
  };
   


  const picker = renderer.create(<AvatarPicker {...exampleData}/>);
  expect(picker).toMatchSnapshot();

});

it('Check for click', () => {
  const exampleData =  {
    imageData: [{
      src: "niceFolder",
      label: 1
    }]
  };
   


  const picker = renderer.create(<AvatarPicker {...exampleData}/>);
  expect(picker).toMatchSnapshot();

});
