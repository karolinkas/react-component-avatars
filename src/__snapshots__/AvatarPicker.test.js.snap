// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check for click 1`] = `
<div
  className="popover"
  onClick={[Function]}
>
  <div>
    <div
      className="gridElement"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <div
        className="image-wrapper"
        onClick={[Function]}
      >
        <img
          alt="avatar"
          className="avatar circle-border "
          src="niceFolder"
        />
        <div
          className="overlay-background"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders without crashing 1`] = `
<div
  className="popover"
  onClick={[Function]}
>
  <div>
    <div
      className="gridElement"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <div
        className="image-wrapper"
        onClick={[Function]}
      >
        <img
          alt="avatar"
          className="avatar circle-border "
          src="niceFolder"
        />
        <div
          className="overlay-background"
        />
      </div>
    </div>
  </div>
</div>
`;
